applications/sentence-transformers-ft/finetune.py

```diff
@@ -0,0 +1,103 @@
+from datasets import load_dataset
+from torch.utils.data import DataLoader
+from sentence_transformers import (
+    SentenceTransformer,
+    InputExample,
+    losses,
+    evaluation,
+    models,
+)
+from torch import nn
+import pathlib
+
+
+# TODO: refactor this to separate finetune from modal code.
+# maybe initialize the dataset inside the function?
+# also maybe make the hyperparameters as parameters of the finetune function
+def finetune(
+    model_id: str,
+    save_path: pathlib.Path,
+    epochs: int = 10,
+    dataset_fraction: int = 1,
+    use_dense_layer: bool = True,
+    dense_out_features: int = 200,
+):
+    """
+    Finetune a sentence transformer on the quora pairs dataset. Evaluates model performance before/after training
+
+    :returns: evaluation accuracy post training
+    :rtype: float
+    Inspired by: https://github.com/UKPLab/sentence-transformers/blob/657da5fe23fe36058cbd9657aec6c7688260dd1f/examples/training/quora_duplicate_questions/training_MultipleNegativesRankingLoss.py
+    """
+
+    # Quora pairs dataset: https://huggingface.co/datasets/quora
+    DATASET_ID = "quora"
+    dataset = load_dataset(DATASET_ID, split="train")
+    # Quora pairs dataset only contains a "train" split in huggingface, so we will manually split it into train and test
+    train_test_split = dataset.train_test_split(test_size=0.1, seed=42)
+    train_dataset = train_test_split["train"]
+    test_dataset = train_test_split["test"]
+
+    if use_dense_layer:
```

Comment:
prosb do something liek 

if dense_out_features:
   ...
   
   saves a variable

Reply:
removed the `use_dense_layer` variable